### ISSUE TITLE
 Call to function "zipmap" failed: number of keys (2) does not match number of values (3).

### ISSUE THREAD

**Issue by Jeanno** [terraform-google-modules/terraform-google-cloud-storage#13]
Summary
===
Run into the following issue when `terraform apply` after removing an item from the `var.names` list
```
Error: Error in function call
  on ../../terraform-google-cloud-storage/outputs.tf line 39, in output "names":
  39:   value       = zipmap(var.names, google_storage_bucket.buckets[*].name)
    |----------------
    | google_storage_bucket.buckets is tuple with 3 elements
    | var.names is list of string with 2 elements
Call to function "zipmap" failed: number of keys (2) does not match number of
values (3).
Error: Error in function call
  on ../../terraform-google-cloud-storage/outputs.tf line 44, in output "urls":
  44:   value       = zipmap(var.names, google_storage_bucket.buckets[*].url)
    |----------------
    | google_storage_bucket.buckets is tuple with 3 elements
    | var.names is list of string with 2 elements
Call to function "zipmap" failed: number of keys (2) does not match number of
values (3).
```

Steps to reproduce
===

1. Use the module with some bucket names (e.g. `names = ["first", "second", "third"]`)
2. `terraform apply`
3. Remove one name from the list (e.g. `names = ["first", "second"]`)
4. `terraform apply` again
4. Encounter the above error

**Comment by ludoo** [comment]
Yup, it's a know issue, workaround is to temporarily comment out the outputs until the new resource is in. Will probably be fixed in a future release, once we switch to using `foreach` instead of `count`.

**Comment by Jeanno** [comment]
So the list ordering issue is also known? The one that if you remove one of the names, the subsequence buckets will be replaced.

**Comment by ludoo** [comment]
Absolutely, that's a Terraform limitation using `count` with resources, and will go away once we switch to `foreach`.

**Comment by Jeanno** [comment]
Any ongoing effort with the change? If not, I might be able to contribute to this.

**Comment by ludoo** [comment]
We did not start yet, as the plan was to do complete 0.12 migration for all the mutiple-resource modules, then switch them all to `foreach`. Migration is done, we just need to find the time to tackle this. If you feel like starting, by all means that is more than welcome. :)

**Comment by morgante** [comment]
FYI I'm not sure this is directly related to `foreach` vs. `count`. It might fix it though.

**Comment by Jberlinsky** [comment]
Just to provide some additional context, this bug has the potential to be quite severe, requiring modifications to Terraform state and manual deletion of buckets it creates to enable Terraform Plans to run in a previously functional configuration.

**Comment by morgante** [comment]
It shouldn't require manual deletion. There's a simple fix anyways.

**Comment by morgante** [comment]
@Jberlinsky The fix is trivial, take a look at #23.
